### PR TITLE
Testing and CI

### PR DESCRIPTION
**Pull request by filmaj**
Checklist of things done:

- [x] ~does ruby and/or python have `npm update` analogues?~ yes they do! added
- [x] the 'slow integration' tests from architect/architect could actually belong ~in this project~ in their own repo, along with the existing filesystem-based tests from this repo (that i've since removed)
- [x] still need to copy the .arc file into events, queues, scheduled and tables
- [x] will need to tweak CI to support all three runtimes. thatll be fun
- [x] a few more unit tests are in order, at least ensuring that certain modules invoke others, i.e. `install` invokes `shared`, that kinda thing
- [x] set up CI on this repo, likely travis
- [x] the readme likely needs a once-over, not sure if it makes sense documenting using `npx` with this module - probably makes more sense to focus on the module-level API
- [x] API _does not_ need to be like the old one. In particular, API will be one-shot for a whole arc project, have default mode hydrate the whole arc app under the assumption execution context is happening from an arc app directory
- [x] support for hydration of all functions, needs to hydrate `shared/` and `views/` and copy shared arc file and static.json file.
- [x] keep the implementation dumb and consistent: execute hydration in series

### CODE REVIEW

**Comment by ryanblock** [comment]
Def into pull relevant slow (aka integration) tests into these repos! (Also what I just did in `sandbox`.)

**Comment by filmaj** [comment]
This one is ready for review, appreciate any comments @brianleroux @ryanblock

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+language: node_js
+node_js:
+  - "8.10"
```
Comment (filmaj) [inline .travis.yml]: We may want to tweak this based on what we decide are minimum node/npm versions needed

File: src/update.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        }
+        if (stderr && stderr.length > 0) {
+          console.log(chalk.yellow(stderr))
+        }
+        callback()
+      }
+
+      // TODO: I think we should consider what minimum version of node/npm this
+      // module needs to use as the npm commands below have different behaviour
+      // depending on npm versio - and enshrine those in the package.json
```
Comment (filmaj) [inline src/update.js]: Def need eyes on this as `npm update`'s behaviour has changed quite a bit from version to version, and I think that should impact what version of node we run against on CI.

**Comment by filmaj** [comment]
Please ignore my plea to review this! Talked with @ryanblock and there's a bunch more work to do here (batch operations w/ a queue / limit, eliminate globing for path identification, maintain old method API for more general usage whether in an arc project or standalone / part of a custom build chain).

Will fold outstanding todos back into the original comment in this PR.

**Comment by filmaj** [comment]
Saw some talk on Slack about possibly wanting to maintain the `glob`ing in this module. Please elaborate / discuss @brianleroux + @ryanblock. Ryan was convinced we should be reading the arc file to figure out which functions need hydrating (and thus `glob`ing shouldn't be used). I am also under the impression the module still needs to be expanded to support hydrating single functions.

**Comment by ryanblock** [comment]
Yeah, that's a pretty good summary. We've been intentionally move Architect's various workflows away from use of globbing files towards more inventory-based (i.e. reading the `.arc` file) operations for quite some time.

Globbing is def a quick and simple way to get the module started, but I don't think it's the right solution to meet Arc's requirements, which will continue to include hydrating single functions's deps when deploying a single function.

In summary:
- We need to be able to hydrate none, one, or many (potentially dozens or hundreds of) functions
- From the .arc file and our existing utils, we already know each of the potential locations for a hydration operations based on Arc pragmas, and what's in the .arc file (i.e. `@http get /foo` → `src/http/get-foo`)
- And we know that if any function defined in `.arc` has a dir with a `package.json`, `Gemfile`, etc. in it, that's what, where, and how we'll hydrate it

**Comment by brianleroux** [comment]
Let's discuss at check-in Monday. Think we need to stay on glob for this function. (Ack that we don't always.)

**Comment by filmaj** [comment]
Assuming we want to keep the `npm update` functionality and its analogues, this one is ready for review.

**Pull request merged**